### ISSUE TITLE
[v6] Dynamic Link buildShortLink return long URL

### ISSUE THREAD

**Issue by kkusanagi** [invertase/react-native-firebase#3282]
Here is the sample code i'm using
```
const url = await dynamiclinks().buildShortLink({
      link:[url],
      domainUriPrefix: 'https://example.page.link',
      ios: {
        bundleId: 'com.example.example',
        appStoreId: '??????',
        fallbackUrl: 'http://example.com',
      },
      android: {
        packageName: 'com.example.example',
        fallbackUrl: 'http://example.com',
      }
    }, 'SHORT');
```
Had tried using 'dynamiclinks.ShortLinkType.SHORT' but got error "Undefined".

Firebase version
    "@react-native-firebase/analytics": "^6.3.4",
    "@react-native-firebase/app": "^6.3.4",
    "@react-native-firebase/auth": "^6.3.4",
    "@react-native-firebase/crashlytics": "^6.3.4",
    "@react-native-firebase/dynamic-links": "^6.3.4",
    "@react-native-firebase/firestore": "^6.3.4",
    "@react-native-firebase/messaging": "^6.3.4",
    "@react-native-firebase/storage": "^6.3.4",
    "react": "16.9.0",
    "react-native": "0.61.5",

**Comment by sandeepsingh91** [comment]
I got the same issue. @kkusanagi  have you find any solution for the same?

**Comment by kkusanagi** [comment]
Nop, I end up downgrade back to v5 react native firebase until all v6 is stable.

**Comment by russellwheatley** [comment]
Hey @kkusanagi , I ran this code:
```
const url = await dynamiclinks().buildShortLink(
      {
        link:
          'https://https://www.google.com/search?q=really+long+url&oq=really+long+url&&sourceid=chrome&ie=UTF-8',
        domainUriPrefix: 'XXXXXX',
        ios: {
          bundleId: 'XXXXXXXX',
          appStoreId: '123456789',
          fallbackUrl: 'https://bbc.com',
        },
      },
      dynamiclinks.ShortLinkType.SHORT,
    );
      console.log(url);
```
`console.log` returned a short url: `https://reactnativefirebase.page.link/oVLi`

You're correct about the static types, I'll be making a PR to include them in the module. 

A couple of things I've noticed about your code. You're setting `link` property as an array when it ought to be a string. You're also using url as the link property and then redeclaring it as a `const` for the short link. Not sure if that is the first time you've declared it without looking at your full code but that might be an issue as well.

**Comment by russellwheatley** [comment]
Here's the merged PR for the static types: #3489

**Comment by kkusanagi** [comment]
Hi, i'm aware of link is String. I just put it that way as short code.
And the "dynamiclinks.ShortLinkType.SHORT" works?
Tried console.log(dynamiclinks.ShortLinkType) and it return "undefined"

**Comment by russellwheatley** [comment]
Sorry, I merged a PR for the static types that will be included in the next release so `dynamiclinks.ShortLinkType` will not work until we release the next version.

**Comment by kkusanagi** [comment]
Noted. And it's ok. Since the v5 still stable and can support my need till now. So this can wait for later upgrade.

**Comment by kkusanagi** [comment]
Hi, I'm going to upgrade to latest version. But i'm not sure had this been fixed? is there a way to check the latest update?

**Comment by Phaen** [comment]
The fix has been merged into master on the 17th of April and included a week later in the release of v6.6.0. If you upgrade to the latest version which as of now is 7.1.0, you should be good.
